osfoffline/database/models.py

```diff
@@ -81,6 +105,14 @@ def validate_top_level(self, key, top_level):
             assert self.parent is not None
         return top_level
 
+    @validates('sync')
+    def validate_sync(self, key, sync):
+        if not self.parent_id:
+            assert sync is True
```

Comment:
Is there a realistic use case where these assertions might fire? We should avoid relying on them for production use- a custom exception class might be more helpful and more robust.

From https://docs.python.org/2/reference/simple_stmts.html#the-assert-statement :
> Assert statements are a convenient way to insert debugging assertions into a program... The current code generator emits no code for an assert statement when optimization is requested at compile time.